app/Libraries/BBCodeFromDB.php

```diff
@@ -382,21 +382,26 @@ public static function removeBlockQuotes($text)
     {
         $level = 0;
         $marker = 0;
+        $quoteBeginning = 0;
 
-        while ($marker >= 0 && $marker < mb_strlen($text) && $level >= 0) {
+        while ($marker >= 0 && $level >= 0) {
             $match = static::scanForNextQuoteTag($text, $marker);
             if ($match === null) {
                 return $text;
             }
 
             if (present($match['start'][0])) {
                 $marker = $match['start'][1] + mb_strlen($match['start'][0]);
+                if ($level === 0) {
+                    $quoteBeginning = $match['start'][1];
+                }
                 $level++;
             } elseif (present($match['end'][0])) {
                 $level--;
                 $marker = $match['end'][1] + mb_strlen($match['end'][0]);
                 if ($level === 0) {
-                    $text = mb_substr($text, $marker, mb_strlen($text) - $marker);
+                    $text = mb_substr($text, 0, $quoteBeginning).mb_substr($text, $marker);
```

Comment:
turns out `preg_match` returns non-multibyte offsets, so we have to use the non `mb_` versions of the string functions or use some combination of the `mb_ereg` functions to get the offsets.

You can test using any string with multibyte characters like
```
ねねぽお誕誕ふお
[quote:1337]abcdef8あうっっとどあえうフォそそcっっ。「」22？？？？[/quote:1337]
茶々ちゃ＼＼・・・＼／as2feff
```